sqlx-cli/src/lib.rs

```diff
@@ -124,26 +122,26 @@ where
 
     let db_url = opts.required_db_url()?;
 
-    backoff::future::retry(
-        backoff::ExponentialBackoffBuilder::new()
-            .with_max_elapsed_time(Some(Duration::from_secs(opts.connect_timeout)))
-            .build(),
-        || {
-            connect(db_url).map_err(|e| -> backoff::Error<anyhow::Error> {
-                if let sqlx::Error::Io(ref ioe) = e {
-                    match ioe.kind() {
-                        io::ErrorKind::ConnectionRefused
-                        | io::ErrorKind::ConnectionReset
-                        | io::ErrorKind::ConnectionAborted => {
-                            return backoff::Error::transient(e.into());
-                        }
-                        _ => (),
-                    }
-                }
+    let (_, v) = {
+        |(mut ctx, db_url): (F, &'a str)| async move {
+            let res = ctx(db_url).await;
+            ((ctx, db_url), res)
+        }
```

Comment:
I'm not a huge fan of the `backon` crate's API design, precisely because it makes more complex calls like this look really goofy.

If the `backoff` crate being unmaintained is an actual issue, the implementation of exponential backoff is simple enough that I'd rather just replace it with something in-tree.

I did start experimenting with my own backoff crate, but it's not published yet and I'm not entirely sure if I'm happy with the API: https://github.com/abonander/ease-off/blob/main/examples/tokio.rs

Reply:
Hi, @abonander, I refined the usage a bit. Does it look better to you now?